### DISCUSSION TITLE
How can I get the GRContext outside render in avalonia 11?

### DISCUSSION THREAD

**Discussion by DrDryg** [AvaloniaUI/Avalonia#13261] [Q&A]
I want to pre-draw some items to (thousands of) SKImage for accelerating my drawing. For that I ideally want my images to reside on the GPU for speed but also possible access to fast compression of textures. For that I need access to the GRContext or create one. I have seen some ways one might do this using AvaloniaLocator but it only applied to pre v11. So how would one go about doing it now for v11?

**Comment by maxkatz6** [comment]
cc @kekekeks

**Reply by kekekeks** [reply]
In general GrContext is considered to be not accessible on the UI thread at all. 
We can probably add some API to access it outside of custom drawing ops / custom visuals, but that needs to involve GPU context loss handling too.

**Reply by DrDryg** [reply]
Ok, then it might not be a good approach. What I want is to have fast drawing (off-screen) in Skia. Do you have some pointer on how I can create my own context? Preferably with the same packages that already are installed with Avalonia.

**Reply by kekekeks** [reply]
You can try enabling private API usage and creating an OpenGL context in the same way it's done in OpenGLControlBase:
https://github.com/AvaloniaUI/Avalonia/blob/43f1b9d8ca9389fea51a5463b003af325b171349/src/Avalonia.OpenGL/Controls/OpenGlControlBase.cs#L71-L114

That would get you your own GPU context that you can feed to Skia.
